pkg/secrets/secrets.go

```diff
@@ -285,7 +303,7 @@ func (sd *SecretDefinition) addSecrets(secretResult *SecretResult) error {
 		return fmt.Errorf("vault listed a secret %s %s, but failed trying to read it; likely the rate-limiting retry attempts were exceeded", keyName, keyPath)
 	}
 
-	if !sd.plural && sd.outputDestination != "" {
+	if !sd.plural && sd.outputDestination != "" && len(sd.secretEnvMapping) == 0 {
```

Comment:
my initial approach was just to remove `sd.outputDestination` but that would make many tests fail, which means that the current API would be broken.

Instead I decided to make this non-breaking change, which doesn't fix the bug, but at least allows for setting environment vairables.